src/tools/clippy/tests/ui/fn_null_check.rs

```diff
@@ -1,22 +0,0 @@
-#![allow(unused)]
-#![warn(clippy::fn_null_check)]
-#![allow(clippy::cmp_null)]
-#![allow(clippy::needless_if)]
-#![allow(clippy::ptr_eq)]
-#![allow(clippy::zero_ptr)]
-
-pub const ZPTR: *const () = 0 as *const _;
-pub const NOT_ZPTR: *const () = 1 as *const _;
-
-fn main() {
-    let fn_ptr = main;
-
-    if (fn_ptr as *mut ()).is_null() {}
-    if (fn_ptr as *const u8).is_null() {}
-    if (fn_ptr as *const ()) == std::ptr::null() {}
-    if (fn_ptr as *const ()) == (0 as *const ()) {}
-    if (fn_ptr as *const ()) == ZPTR {}
-
-    // no lint
-    if (fn_ptr as *const ()) == NOT_ZPTR {}
```

Comment:
you lost these tests.

Reply:
that's because unlike clippy we don't lint on them

linting on them would require a similar infrastructure as [clippy_utils::consts::constant](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/consts/fn.constant.html), which evaluates HIR expression and as far as I can see rustc does not have an equivalent; one could probably be brought but I don't think it's worth it

Reply:
Yes, but we should still have tests for them. Especially if we explicitly decided against it.

Reply:
also the clippy expression folder is for `1 - 1` producing `0`. In this case you could in theory just evaluate the constant, but you're right, we should not make this more complicated